tests/oss-fuzz/build.sh

```diff
@@ -36,21 +36,36 @@
 #     avif_fuzztest_enc_dec_incr@EncodeDecodeAvifFuzzTest.EncodeDecodeGridValid \
 #     --sanitizer address
 
-# Reset compile flags to build dav1d without fuzzer flags. The meson build system
-# is problematic with sanitizer flags.
+# Reset compile flags to build libyuv without fuzzer flags.
 export ORIG_CFLAGS="$CFLAGS"
 export ORIG_CXXFLAGS="$CXXFLAGS"
 export CFLAGS=""
 export CXXFLAGS=""
 
-cd ext && bash dav1d.cmd && bash libyuv.cmd && cd ..
+cd ext && bash libyuv.cmd && cd ..
 
 export CFLAGS=$ORIG_CFLAGS
 export CXXFLAGS=$ORIG_CXXFLAGS
 
+# Build dav1d with sanitizer flags.
+# Adds extra flags: -Db_sanitize=$SANITIZER -Db_lundef=false, and -Denable_asm=false for msan
```

Comment:
If the `-Db_lundef=false` flag is necessary for only certain sanitizers, it would be good to specifically check for those sanitizers.

But when I searched for the error message "undefined reference to `__afl_area_ptr'" on the Web and looked at the results, I think this linker error is caused by compiling dav1d with afl-clang-fast and compiling libavif with clang.

afl-clang-fast comes from the oss-fuzz environment:

```
Step #3 - "compile-afl-address-x86_64": ---------------------------------------------------------------
Step #3 - "compile-afl-address-x86_64": CC=/src/aflplusplus/afl-clang-fast
Step #3 - "compile-afl-address-x86_64": CXX=/src/aflplusplus/afl-clang-fast++
Step #3 - "compile-afl-address-x86_64": CFLAGS=-O1 -fno-omit-frame-pointer -gline-tables-only -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION -fsanitize=address -fsanitize-address-use-after-scope 
Step #3 - "compile-afl-address-x86_64": CXXFLAGS=-O1 -fno-omit-frame-pointer -gline-tables-only -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION -fsanitize=address -fsanitize-address-use-after-scope  -stdlib=libc++
Step #3 - "compile-afl-address-x86_64": RUSTFLAGS=--cfg fuzzing -Zsanitizer=address -Cdebuginfo=1 -Cforce-frame-pointers
Step #3 - "compile-afl-address-x86_64": ---------------------------------------------------------------
```

while clang comes from the `-DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++` cmake options at line 80.

Could you try removing `-DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++` and removing `-Db_lundef=false`?

Reply:
Vincent: I tested the changes I suggested locally. They passed the `check_build` step. To be safe, I will make the changes in two steps.

Here is the first pull request to remove `-DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++` (https://github.com/AOMediaCodec/libavif/pull/2028), which I am confident is correct (unless I miss something subtle).

Reply:
Indeed, it is now fine